src/textual/_config.py

```diff
@@ -0,0 +1,207 @@
+from __future__ import annotations
+
+from pathlib import PurePath
+from typing import Iterable, Collection
+
+import tomli
+
+
+class Config:
+    def __init__(
+        self,
+        namespace: str,
+        default_config_path: str | PurePath,
+        user_config_paths: Iterable[str | PurePath] = None,
+    ):
+        """Manages access to configuration for the Textual application.
+        Textual applications may ship with a `textual.toml` file containing default configuration values.
+        We refer to these as "packaged defaults", and they may only contain relating to the Textual app.
+        Users may have a `textual.toml` file on their system which may override these default configuration values.
+        Instantiating this class does not result in disk access. Disk access only occurs when the `resolve` method is
+        called.
+
+        Args:
+            namespace (str): The namespace of this application. Used to filter out irrelevant config.
+            default_config_path (str | PurePath): The path to the default config for this app. This type of config is
+                handled separately from user configuration as it's more restrictive. For example, you cannot bundle
+                a config file with your application that alters global configuration.
+            user_config_paths (Iterable[str | PurePath]): The paths to the config files to load on the users machine.
+                If items appear multiple times in the supplied config files and have matching namespaces, the last
+                occurrence will take priority.
+        """
+        self.namespace = namespace
+        self._default_config_path = default_config_path
+        self._user_config_paths = user_config_paths
+        self._raw_merged_config = {}
+
+        # Textual default config values which can be overridden by resolved config.
+
+        # TODO: Apply environment variables first before resolving configuration files
+        self.dark = False
+        self.debug = False
+
+    def __getitem__(self, key: str) -> object:
+        """Retrieve a config value for this application for a given key."""
+        return self._raw_merged_config.get(self.namespace)[key]
+
+    @property
+    def raw(self) -> dict[str, object]:
+        return self._raw_merged_config.get(self.namespace)
+
+    @property
+    def meta(self) -> dict[str, object]:
+        return self._raw_merged_config.get("meta")
```

Comment:
It would contain the namespace stuff rather than having it at a global scope (which is trickier to manage). I think you're probably right. When things are a little more solidified I'll scrap this section if needs be.